server/sonar-server/src/main/java/org/sonar/server/component/ws/FilterParser.java

```diff
@@ -29,11 +30,16 @@
 import javax.annotation.Nullable;
 import org.sonar.core.util.stream.Collectors;
 
+import static java.util.Objects.requireNonNull;
+
 public class FilterParser {
 
   private static final Splitter CRITERIA_SPLITTER = Splitter.on(Pattern.compile("and", Pattern.CASE_INSENSITIVE));
-  private static final Pattern TERNARY_PATTERN = Pattern.compile("(\\w+)\\s+(\\S+)\\s+(\\w+)");
-  private static final Pattern SINGLE_KEY_PATTERN = Pattern.compile("(\\w+)");
+  private static final Splitter IN_VALUES_SPLITTER = Splitter.on(",").omitEmptyStrings().trimResults();
+
+  private static final Pattern PATTERN_HAVING_VALUE = Pattern.compile("(\\w+)\\s+(\\S+)\\s+(\\w+)");
```

Comment:
We could, but as it will be probably refactored later, I prefer to postpone this refactoring